### PR TITLE
clean v2 docs

### PR DESCRIPTION
**Pull request by williamFalcon**
Improves on v1 of new docs started by @Borda 
https://torch-lightning.readthedocs.io/en/paths/documentation.html

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,8 @@ tests/save_dir
 default/
 lightning_logs/
 tests/tests/
+*.rst
```
Comment (Borda) [inline .gitignore]: I would rather write
```
pl_examples.rst
pl_examples.*.rst
pytorch_lightning.rst
pytorch_lightning.*.rst
```

**Comment by Borda** [comment]
Pls could you rebase it on #685 since this is a kind of follow-up and check #687

File: pytorch_lightning/callbacks/pt_callbacks.py [hunk 2]
```diff
@@ -150,20 +160,21 @@ def on_train_end(self, logs=None):
...
-    then the model checkpoints will be saved with the epoch number and
-    the validation loss in the filename.
-
-    # Arguments
-        filepath: string, path to save the model file.
-        monitor: quantity to monitor.
-        verbose: verbosity mode, 0 or 1.
-        save_top_k: if `save_top_k == k`,
+    r"""
+
```
Comment (Borda) [inline pytorch_lightning/callbacks/pt_callbacks.py]: rather start just after """

File: pytorch_lightning/callbacks/pt_callbacks.py [hunk 3]
```diff
@@ -62,25 +66,31 @@ def on_train_end(self, logs=None):
...
+        mode (str): one of {auto, min, max}. In `min` mode,
             training will stop when the quantity
             monitored has stopped decreasing; in `max`
             mode it will stop when the quantity
             monitored has stopped increasing; in `auto`
             mode, the direction is automatically inferred
             from the name of the monitored quantity.
-
+    Example:
+        >>> from pytorch_lightning import Trainer
```
Comment (Borda) [inline pytorch_lightning/callbacks/pt_callbacks.py]: pls remove >>> or keep it and shift left so it become a doctest

File: pytorch_lightning/callbacks/pt_callbacks.py [hunk 4]
```diff
@@ -150,20 +160,21 @@ def on_train_end(self, logs=None):
...
+
+    Args:
+        filepath (str): path to save the model file.
+            Can contain named formatting options to be auto-filled.
+
+            Example:
+                >>> # save epoch and val_loss in name
+                >>> ModelCheckpoint(filepath='{epoch:02d}-{val_loss:.2f}.hdf5')
+                >>> # saves file like: /path/epoch_2-val_loss_0.2.hdf5
+        monitor (str): quantity to monitor.
```
Comment (Borda) [inline pytorch_lightning/callbacks/pt_callbacks.py]: We shall decide what kind of documentation we are going to use...
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: pytorch_lightning/core/lightning.py [hunk 5]
```diff
@@ -1,3 +1,14 @@
+"""
+The LightningModule is the "system recipe." It groups the following in one file:
+    - computational system definition
```
Comment (Borda) [inline pytorch_lightning/core/lightning.py]: I think that it needs a blank line to be treated as a list, be aware that reStructuredText and MarkDown formatting is a bit different...
http://www.unexpected-vortices.com/doc-notes/markdown-and-rest-compared.html

File: pytorch_lightning/trainer/trainer.py [hunk 6]
```diff
@@ -94,15 +94,63 @@ def __init__(
...
+                        prefix=''
+                    )
+
+                    trainer = Trainer(checkpoint_callback=checkpoint_callback)
             early_stop_callback (:class:`.EarlyStopping`): Callback for early stopping
-            default_save_path (str): Default path for logs+weights if no logger/ckpt_callback passed
-            gradient_clip_val (int): 0 means don't clip.
-            gradient_clip (int): 0 means don't clip. Deprecated.
-            process_position (int): shown in the tqdm bar
+                Example::
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py]: just a reminder, it still will be on the same line as "early_stop_callback (:class:`.EarlyStopping`): Callback for early stopping Example"

Comment (williamFalcon) [reply pytorch_lightning/trainer/trainer.py]: i would wait a bit until this is more done hahaha. this will continue to change :)

**Comment by williamFalcon** [comment]
@Borda ready for review!
These aren't 100% where I think they should be but I propose the following:

1. merge asap
2. deploy to docs url
3. work on examples, tutorials and common use cases over the next few days. Some of those are because I'm not super familiar with this formatting for those sections.

**Pull request merged**